From c3c65c199862f2f4c910a0e019e4a04f3311f8b1 Mon Sep 17 00:00:00 2001
From: misfo <tedwardo2@gmail.com>
Date: Thu, 12 Mar 2009 07:25:07 -0500
Subject: [PATCH] allowed render :file to take Pathnames

This is useful for doing something like this:
render :file => Rails.root.join("public", "meaning_of_life.html")
---
 actionpack/lib/action_view/template.rb    |    2 +-
 actionpack/test/controller/render_test.rb |   10 ++++++++++
 2 files changed, 11 insertions(+), 1 deletions(-)

diff --git a/actionpack/lib/action_view/template.rb b/actionpack/lib/action_view/template.rb
index 0dd3a7e..c339c8a 100644
--- a/actionpack/lib/action_view/template.rb
+++ b/actionpack/lib/action_view/template.rb
@@ -218,7 +218,7 @@ module ActionView #:nodoc:
       # Returns file split into an array
       #   [base_path, name, locale, format, extension]
       def split(file)
-        if m = file.match(/^(.*\/)?([^\.]+)\.(.*)$/)
+        if m = file.to_s.match(/^(.*\/)?([^\.]+)\.(.*)$/)
           base_path = m[1]
           name = m[2]
           extensions = m[3]
diff --git a/actionpack/test/controller/render_test.rb b/actionpack/test/controller/render_test.rb
index af62339..a529315 100644
--- a/actionpack/test/controller/render_test.rb
+++ b/actionpack/test/controller/render_test.rb
@@ -157,6 +157,11 @@ class TestController < ActionController::Base
     render :file => 'test/dot.directory/render_file_with_ivar'
   end
 
+  def render_file_using_pathname
+    @secret = 'in the sauce'
+    render :file => Pathname.new(File.dirname(__FILE__)).join('..', 'fixtures', 'test', 'dot.directory', 'render_file_with_ivar.erb')
+  end
+
   def render_file_from_template
     @secret = 'in the sauce'
     @path = File.expand_path(File.join(File.dirname(__FILE__), '../fixtures/test/render_file_with_ivar.erb'))
@@ -861,6 +866,11 @@ class RenderTest < ActionController::TestCase
     assert_equal "The secret is in the sauce\n", @response.body
   end
 
+  def test_render_file_using_pathname
+    get :render_file_using_pathname
+    assert_equal "The secret is in the sauce\n", @response.body
+  end
+
   def test_render_file_with_locals
     get :render_file_with_locals
     assert_equal "The secret is in the sauce\n", @response.body
-- 
1.6.1.2

